### ISSUE TITLE
Multiple-root workspaces broken

### ISSUE THREAD

**Issue by nwolverson** [nwolverson/vscode-ide-purescript#104]
- Commands won't work on additional workspace folders, and in fact the attempt to register commands means the initialisation of the additional LS fails
- Command will need to target correct workspace
- Additional LS startup seems to be for the wrong folder

How did this ever seem to work? (Does anybody use multi-root?)

**Comment by starper** [comment]
I was trying to use multi-root yesterday, but I gave up pretty fast :) Two things that I noticed:

1. There is no way to use `purescript` settings on folder level, only on workspace level.
2. `Start`/`Stop`/`Restart` commands do not work as expected. It seems like they work only for the server that was launched first, others are ignored.

**Comment by nwolverson** [comment]
The 2nd is just the commands issue above.

The 1st I couldn't find docs on before, but it looks like we need to apply settings at the `resource` level and this could work.

**Comment by spicydonuts** [comment]
@nwolverson Do you know if including multiple src or package paths is possible? I tried an include string which works in `pscid`, like `pscid -I 'path1;path2;path3'`, and a few others but haven't gotten it to work yet.

**Comment by nwolverson** [comment]
No, don't think so, these are used to construct globs just now. Could be `{path1,path2}` could work

**Comment by spicydonuts** [comment]
It doesn't seem to

**Comment by jamesdbrock** [comment]
Instead of using `sourcePath` in `settings.json`, you can use `PURS_IDE_SOURCES` in your dev environment. Example:

```
nixpkgs.mkShell {
  PURS_IDE_SOURCES = "'src/**/*.purs plugin/**/*.purs test/**/*.purs .spago/**/*.purs'";
}
```

Reference https://github.com/nwolverson/purescript-language-server/pull/75

**Comment by nwolverson** [comment]
Note that if you use `PURS_IDE_SOURCES` you will no longer have the correct dependency source globs from spago, so you may need to manually delete old hashes that are no longer used
